scripts/source-install-preflight.sh

```
@@ -0,0 +1,268 @@
+
+case "${MODE}" in
+    check|claim|ensure-dir|publish-cli|publish-gateway) ;;
+    *) usage ;;
+esac
+
+[[ -d "${REPO_ROOT_INPUT}" ]] || {
+    echo "error: source-install repository root does not exist: ${REPO_ROOT_INPUT}" >&2
+    exit 64
+}
+
+readonly REPO_ROOT="$(cd "${REPO_ROOT_INPUT}" && pwd -P)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)